lib/oli_web/live/delivery/open_and_free_index.ex

```diff
@@ -0,0 +1,179 @@
+            <% else %>
+              <div class="flex flex-wrap">
+                <.link
+                  :for={section <- @filtered_sections}
+                  href={~p"/sections/#{section.slug}/overview"}
+                  class="rounded-lg shadow-lg bg-white dark:bg-gray-600 max-w-xs mr-3 mb-3 border-2 border-transparent hover:border-blue-500 hover:no-underline"
+                >
+                  <img
+                    class="rounded-t-lg object-cover h-64 w-96"
+                    src={cover_image(section)}
+                    alt="course image"
+                  />
+                  <span class="badge badge-info ml-2 mt-2 capitalize"><%= section.user_role %></span>
+                  <div class="p-6">
+                    <h5 class="text-gray-900 dark:text-white text-xl font-medium mb-2">
+                      <%= section.title %>
+                    </h5>
+                    <p class="text-gray-700 dark:text-white text-base mb-4">
+                      <%= section.description %>
+                    </p>
+                  </div>
+                </.link>
+                <p :if={length(@filtered_sections) == 0} class="mt-4">
+                  No course found matching <strong>"<%= @params.text_search %>"</strong>
+                </p>
+              </div>
+            <% end %>
+          </div>
+        </div>
+      </div>
+    </main>
+    <%= render(OliWeb.LayoutView, "_delivery_footer.html", assigns) %>
+    """
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("search_section", %{"search" => text_search}, socket) do
+    {:noreply, push_patch(socket, to: ~p"/sections?#{%{text_search: text_search}}")}
+  end
+
+  defp add_user_role([], _user), do: []
+
+  defp add_user_role(sections, user) do
+    sections
+    |> Enum.map(fn s ->
+      Map.merge(s, %{user_role: Utils.user_role(s, user) |> Atom.to_string()})
+    end)
+  end
+
+  defp add_instructors([]), do: []
+
+  defp add_instructors(sections) do
+    instructors_per_section = instructors_per_section(sections)
+
+    sections
+    |> Enum.map(fn section ->
+      Map.merge(section, %{instructors: Map.get(instructors_per_section, section.id, [])})
+    end)
+  end
+
+  defp instructors_per_section(sections) do
+    section_ids = Enum.map(sections, & &1.id)
+    instructor_context_role_id = ContextRoles.get_role(:context_instructor).id
+
+    query =
+      from(
+        e in Enrollment,
+        join: s in Section,
+        on: e.section_id == s.id,
+        join: ecr in EnrollmentContextRole,
+        on: e.id == ecr.enrollment_id,
+        where:
+          s.id in ^section_ids and s.status == :active and e.status == :enrolled and
+            ecr.context_role_id == ^instructor_context_role_id,
+        preload: [:user],
+        select: {s.id, e}
+      )
+
+    Repo.all(query)
+    |> Enum.group_by(fn {section_id, _} -> section_id end, fn {_, enrollment} ->
+      Utils.user_name(enrollment.user)
+    end)
```

Comment:
WDYT about moving this query to the `lib/oli/delivery/sections.ex` context file?